config/metrics/crd-metrics-config.yaml

```
@@ -0,0 +1,676 @@
+            - status
+            type:
+            - type
+          nilIsZero: false
+          path:
+          - status
+          - conditions
+          valueFrom:
+          - lastTransitionTime
+        type: Gauge
+      help: The condition last transition time of a vspheredeploymentzone.
+      name: status_condition_last_transition_time
+    resourcePlural: ""
+  - errorLogV: 0
+    groupVersionKind:
+      group: infrastructure.cluster.x-k8s.io
+      kind: VSphereFailureDomain
+      version: v1beta1
+    labelsFromPath:
+      cluster_name:
+      - metadata
+      - labels
+      - cluster.x-k8s.io/cluster-name
+      name:
+      - metadata
+      - name
+      namespace:
+      - metadata
+      - namespace
+      uid:
+      - metadata
+      - uid
+    metricNamePrefix: capv_vspherefailuredomain
+    metrics:
+    - each:
+        info:
+          labelFromKey: ""
+          labelsFromPath:
+            paused_value: []
+          path:
+          - metadata
+          - annotations
+          - cluster.x-k8s.io/paused
+        type: Info
+      help: Whether the vspherefailuredomain is paused and any of its resources will
+        not be processed by the controllers.
+      name: annotation_paused
+    - each:
+        gauge:
+          labelFromKey: ""
+          nilIsZero: false
+          path:
+          - metadata
+          - creationTimestamp
+          valueFrom: null
+        type: Gauge
+      help: Unix creation timestamp.
+      name: created
+    - each:
+        info:
+          labelFromKey: ""
+          labelsFromPath:
+            name:
+            - metadata
+            - name
+          path: null
+        type: Info
+      help: Information about a vspherefailuredomain.
+      name: info
+    - each:
+        info:
+          labelFromKey: ""
+          labelsFromPath:
+            owner_is_controller:
+            - controller
+            owner_kind:
+            - kind
+            owner_name:
+            - name
+            owner_uid:
+            - uid
+          path:
+          - metadata
+          - ownerReferences
+        type: Info
+      help: Owner references.
+      name: owner
+    resourcePlural: ""
+  - errorLogV: 0
+    groupVersionKind:
+      group: infrastructure.cluster.x-k8s.io
+      kind: VSphereMachine
+      version: v1beta1
+    labelsFromPath:
+      cluster_name:
+      - metadata
+      - labels
+      - cluster.x-k8s.io/cluster-name
+      name:
+      - metadata
+      - name
+      namespace:
+      - metadata
+      - namespace
+      uid:
+      - metadata
+      - uid
+    metricNamePrefix: capv_vspheremachine
+    metrics:
+    - each:
+        info:
+          labelFromKey: ""
+          labelsFromPath:
+            address:
+            - address
+            type:
+            - type
+          path:
+          - status
+          - addresses
+        type: Info
+      help: Address information about a vspheremachine.
+      name: addresses
```

```suggestion
      name: status_addresses
```
Same in core CAPI I think

---

Same for VSphereVM